Update SSL docs with FIPS 140-3 (#3087)
neo4j/docs-operations: modules/ROOT/pages/security/ssl-fips-compatibility.adoc

Author: When do we want to publish this update?

[[ssl-fips-compatibility]]
= Configuring SSL for FIPS 140-3 compatibility
:description: How to configure Neo4j to use FIPS compatible SSL encryption.
:keywords: ssl, tls, authentication, encryption, encrypted, security, fips, fips 140, fips 140-2, fips 140-3, nist, hipaa
:page-role: enterprise-edition

Federal Information Processing Standards (FIPS) 140 is a U.S. government standard established by the National Institute of Standards and Technology (NIST) which is used to accredit cryptographic modules such as those used in TLS network encryption.
Although FIPS 140 compliance is primarily required for federal agencies and their contractors, it is also used in the healthcare sector under regulations like the Health Insurance Portability and Accountability Act (HIPAA) to protect patient data.

This guide helps configure Neo4j to use TLS/SSL encryption in a FIPS-compliant way.
It is supplementary to the xref:security/ssl-framework.adoc[] documentation, as many of the configuration processes and requirements are the same.

[NOTE]
====

> FIPS 140-2 will reach EOL on September 21, 2026, and all the FIPS 140-2 certificates will be placed on the historical list.

Author: ```suggestion
FIPS 140-2 will reach end of life (EOL) on September 21, 2026.
After this date, all the FIPS 140-2 certificates will be moved to the historical list.
```

You should audit your active systems and migrate to FIPS 140-3 validated alternatives.

> All Neo4j 2026.x versions support FIPS 140-3 compatible SSL encryption.

Author: To double-check: only 2026.x versions? 2025.x versions do not support FIPS 140-3?

> All Neo4j 2026.x versions support FIPS 140-3 compatible SSL encryption.

Author: ```suggestion
All Neo4j 2025.x and 2026.x versions support FIPS 140-3 compatible SSL encryption.
```

> All Neo4j 2025.x and 2026.x versions support FIPS 140-3 compatible SSL encryption.

Author: ```suggestion
All Neo4j 2025.x and 2026.x versions support SSL encryption compatible with FIPS 140-3 requirements.
```
